### PR TITLE
(ACTION) Add smart enter action

### PR DESCRIPTION
**Pull request by alexpana**
The smart enter action applies a series of fixes in order to complete the current line before actually inserting a new line and moving the carent on the next line.

This commit adds simple fixers for closing function call parenthesis and adding a semicolon.

### CODE REVIEW

**Comment by alexpana** [comment]
This is a fix for #247

File: src/main/kotlin/org/rust/ide/actions/smartenter/SemicolonFixer.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class SemicolonFixer : SmartEnterProcessorWithFixers.Fixer<RustSmartEnterProcessor>() {
+    override fun apply(editor: Editor, processor: RustSmartEnterProcessor, element: PsiElement) {
+        if (isOutermostCallExpr(element) && !endsWithSemicolon(element)) {
+            editor.document.insertString(element.textRange.endOffset, ";");
+        }
+    }
+
+    private fun isOutermostCallExpr(element: PsiElement): Boolean {
+        return element is RustCallExpr && PsiTreeUtil.getParentOfType(element, RustCallExpr::class.java) == null;
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/actions/smartenter/SemicolonFixer.kt]: We use Kotlin's [Reified type parameters](https://kotlinlang.org/docs/reference/inline-functions.html#reified-type-parameters) to provide a more Kotlinish API for PsiTreeUtil [here](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/lang/core/psi/util/RustPsiExtensions.kt#L19).

You can write just `element.getParentOfType<RustCallExpr>`.

File: src/main/kotlin/org/rust/ide/actions/smartenter/SemicolonFixer.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    private fun endsWithSemicolon(element: PsiElement): Boolean {
+        if (element.nextSibling != null) {
+            if (element.nextSibling is PsiWhiteSpace) {
+                return endsWithSemicolon(element.nextSibling);
+            }
+            return element.nextSibling.text == ";";
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/actions/smartenter/SemicolonFixer.kt]: Ah, and you don't need semicolons in Kotlin :)

**Comment by matklad** [comment]
Nice tests!

**Comment by matklad** [comment]
Thanks!

**Pull request merged**